code/modules/mob/living/carbon/human/human_powers.dm

```diff
@@ -1336,3 +1336,837 @@
+			remoteview_target = target
+			reset_view(target)
+			return
+		if(choice == "Deny")
+			if(istype(src.origin, /singleton/origin_item/origin/mouv_b))
+				var/hijack = alert(src, "[target] has denied your access request. Attempt a hijack?", "Hijack Hivenet Senses", "No", "Yes")
+				if(hijack == "Yes")
+					if(prob(40) && !S.shielded)
+						to_chat(src, SPAN_NOTICE("You extend your mind into the Hivenet, accessing [target]'s senses. Use this verb again to cancel."))
+						remoteview_target = target
+						reset_view(target)
+						return
+					else
+						to_chat(src, SPAN_WARNING("You attempt a hijack, and a stab of pain comes to your mind as your attempt is rejected!"))
+						src.adjustHalLoss(15)
+						src.flash_pain(15)
+						to_chat(target, SPAN_WARNING("[src] attempted to hijack your senses, but your countermeasures repelled them!"))
+						remoteview_target = null
+						reset_view(0)
+						host.last_action = world.time + 5 MINUTES
+						return
+		to_chat(src, SPAN_NOTICE("Your request to access [target]'s senses has been denied."))
+		remoteview_target = null
+		reset_view(0)
+
+/mob/living/carbon/human/proc/hivenet_neuralshock()
+	set name = "Neural Shock"
+	set desc = "Shock a member of your own Hive with a functioning Neural Socket. Deals pain and brain damage. Rare to need to use, most Ta would only do so in the event of a malfunctioning Vaurca or Viax. A Mouv Ta may attempt this against other Hive members, though to do so would be an extreme diplomatic breach."
+	set category = "Hivenet"
+
+	var/list/available_vaurca = list()
+	var/list/fullname = splittext(src.name, " ")
+	var/surname = fullname[2]
+	var/obj/item/organ/internal/vaurca/neuralsocket/host = src.internal_organs_by_name[BP_NEURAL_SOCKET]
+	if(src.stat != CONSCIOUS)
+		to_chat(src, SPAN_WARNING("You are incapable of that in your current state!"))
+		return
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages) || !host)
+		to_chat(src, "<span class='danger'>Your mind is dark, unable to communicate with the Hive.</span>")
+		return
+	if(!istype(host, /obj/item/organ/internal/vaurca/neuralsocket/admin))
+		to_chat(src, "<span class='danger'>You do not have the authority to do this!.</span>")
+		return
+	if(host.disrupted || host.muted)
+		to_chat(src, SPAN_WARNING("Your neural socket is unable to broadcast!"))
+		return
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You must wait before doing this!"))
+		return
+	for(var/mob/living/carbon/human/player in living_mob_list)
+		if(isvaurca(player) && player.internal_organs_by_name[BP_NEURAL_SOCKET])
+			var/list/player_fullname = splittext(player.name, " ")
+			var/player_surname = player_fullname[2]
+			if(player_surname == surname || istype(src.origin, /singleton/origin_item/origin/mouv_b))
+				LAZYADD(available_vaurca, player)
+	LAZYREMOVE(available_vaurca, src) //can't shock yourself
```

Comment:
not a lazy list

also this codeblock is repeated a lot, maybe make it into a proc